### ISSUE TITLE
[clang-tidy] false positive for misc-redundant-expression with fold-expression

### ISSUE THREAD

**Issue by schenker** [llvm/llvm-project#110089]
In some cases, clang-tidy reports a false-positive misc-redundant-expression for fold expressions that contain the same value more than once (https://godbolt.org/z/YfThnE9Wh):
```cpp
#include <type_traits>

template<typename... Args>
constexpr bool all_true = (... && Args::value);
constexpr bool result = all_true<std::true_type, std::true_type, std::true_type>;
```

Output:
```
<source>:4:28: warning: both sides of operator are equivalent [misc-redundant-expression]
    4 | constexpr bool all_true = (... && Args::value);
      |           
```
Strictly speaking the warning is correct, but I would not expect clang-tidy to complain here.
 
When `all_true` is implemented as a function, clang-tidy does not report anything:
```cpp
#include <type_traits>

template<typename... Args>
constexpr bool all_true() {
    return (... && Args::value);
}

constexpr bool result = all_true<std::true_type, std::true_type, std::true_type>();
```

**Comment by schenker** [comment]
It appears that this issue has been resolved, and I can no longer reproduce the problem. I will close this issue.
